pkg/operator/ceph/csi/peermap/config.go

```diff
@@ -0,0 +1,337 @@
+		}
+
+		logger.Debugf("Pool details of local cluster %+v", localPoolDetails)
+
+		for _, peerSecret := range pool.Spec.Mirroring.Peers.SecretNames {
+			s, err := clusterContext.Clientset.CoreV1().Secrets(clusterInfo.Namespace).Get(context.TODO(), peerSecret, metav1.GetOptions{})
+			if err != nil {
+				return mappings, errors.Wrapf(err, "failed to fetch kubernetes secret %q bootstrap peer", peerSecret)
+			}
+
+			token := s.Data["token"]
+			decodedTokenToGo, err := decodePeerToken(string(token))
+			if err != nil {
+				return mappings, errors.Wrap(err, "failed to decode bootstrap peer token")
+			}
+
+			peerClientName := fmt.Sprintf("client.%s", decodedTokenToGo.ClientID)
+			credentials := cephclient.CephCred{
+				Username: peerClientName,
+				Secret:   decodedTokenToGo.Key,
+			}
+
+			// Add cluster ID mappings
+			clusterIDMapping := map[string]string{
+				decodedTokenToGo.Namespace: clusterInfo.Namespace,
+			}
+
+			mappings.addClusterIDMapping(clusterIDMapping)
+
+			// Generate peer cluster keyring in a temporary file
+			keyring := cephclient.CephKeyring(credentials)
+			keyringFile, err := createTmpFile(keyring, "tmp-keyring")
+			if err != nil {
+				return mappings, errors.Wrap(err, "failed to create a temp keyring file")
+			}
+			defer removeTmpFile(keyringFile.Name())
+
+			// Generate an empty config file to be passed as `--conf`argument in ceph CLI
+			configFile, err := createTmpFile("", "tmp-config")
+			if err != nil {
+				return mappings, errors.Wrap(err, "failed to create a temp config file")
+			}
+			defer removeTmpFile(configFile.Name())
+
+			// Build command
+			args := []string{"osd", "pool", "get", pool.Name, "all",
+				fmt.Sprintf("--cluster=%s", decodedTokenToGo.Namespace),
+				fmt.Sprintf("--conf=%s", configFile.Name()),
+				fmt.Sprintf("--fsid=%s", decodedTokenToGo.ClusterFSID),
+				fmt.Sprintf("--mon-host=%s", decodedTokenToGo.MonHost),
+				fmt.Sprintf("--keyring=%s", keyringFile.Name()),
+				fmt.Sprintf("--name=%s", peerClientName),
+				"--format", "json",
+			}
+
+			// Get peer cluster pool details
+			peerPoolDetails, err := getPeerPoolDetails(clusterContext, args...)
+			if err != nil {
+				return mappings, errors.Wrapf(err, "failed to get pool details from peer cluster %q", decodedTokenToGo.Namespace)
+			}
+
+			logger.Debugf("Pool details from peer cluster %+v", peerPoolDetails)
+
+			// Add Pool ID mappings
+			poolIDMapping := map[string]string{
+				strconv.Itoa(peerPoolDetails.Number): strconv.Itoa(localPoolDetails.Number),
+			}
+
+			mappings.addRBDPoolIDMapping(clusterIDMapping, poolIDMapping)
+		}
+	}
+
+	return mappings, nil
+}
+
+func CreateOrUpdateConfig(
+	clusterContext *clusterd.Context,
+	mappings *PeerIDMappings,
+) error {
+	ctx := context.TODO()
+	data, err := mappings.toString()
+	if err != nil {
+		return errors.Wrap(err, "failed to convert peer cluster mappings struct to string")
+	}
+
+	opNamespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
+	if opNamespace == "" {
```

Comment:
This error is not reachable since the operator validates this in the init sequence. So this is not needed